050-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,744 @@
+>       finalizer: 
+>       - strimzi.io/topic-operator
+>     spec:
+>       # ...
+>     status:
+>       observedGeneration: 123 # == generation => it is Kube-managed
+>     ```
+> 
+> 2. The user changes the `strimzi.io/managed=false` annotation to `false` so that the topic is no longer Kube-managed.
+> 
+>     ```yaml=
+>     metadata:
+>       generation: 124
+>       name: foo
+>       finalizer: 
+>       - strimzi.io/topic-operator
+>       annotations:
+>       - strimzi.io/managed: false # do not synchronize to Kafka
+>     spec:
+>       # ...
+>     status:
+>       observedGeneration: 123 # != generation => it is Kube-managed
+>     ```
+> 
+> 
+> 3. The user can then delete the resource.
+>     Because of the presence of a `metadata.finalizer` the resource is not actually deleted.
+>     Instead Kube adds the `metadata.deletionTimestamp` field.
+> 
+>     ```yaml=
+>     metadata:
+>       generation: 124
+>       name: foo
+>       finalizer: 
+>       - strimzi.io/topic-operator
+>       annotations:
+>       - strimzi.io/managed: false # do not synchronize to Kafka
+>       deletionTimestamp: 20230301T000000.000
+
+>     spec:
+>       # ...
+>     status:
+>       observedGeneration: 124 # == generation => it is not syncrhonized
+>     ```
+> 
+> 4. The operator notices the `metadata.deletionTimestamp` field.
+>     Since the `spec.magnaged` is `False` it does **not** attempt topic deletion.
+>     1. It removes `strimzi.io/topic-operator` from the `metadata.finalizer`.
+>     2. Kube proceeds to remove the resource.
+
+Note that `strimzi.io/managed: false` also means that a `KafkaTopic` can exist without there being a corresponding topic in Kafka.
+
+---
+
+
+### Operational considerations
+
+#### Changing `metadata.name`
+
+This WILL be supported by:
+
+1. unmanaging the topic (`strimzi.io/managed: false`).
+2. deleting the `KafkaTopic`.
+3. recreating the `KafkaTopic` with a new name.
+
+#### Change the `metadata.namespace` 
+
+In the case of a UTO configured for multiple namespaces it will be possible to move the `KafkaTopic` for a topic in Kafka from one namespace to another by:
+
+  1. Unmanaging the `KafkaTopic` in the old namespace (as described above)
+  2. Deleting the `KafkaTopic` from the old namespace,
+  3. Revising the policy (requiring a UTO restart)
+  4. Creating a new `KafkaTopic` in the new namespace.
+
+#### Changing `spec.topicName`
+
+This will NOT be supported.
+It will be detected by a mismatch between `spec.topicName` and `status.topicName`.
+Changes thus detected will result in an error condition.
+
+#### Decreasing `spec.partitions`
+
+This will NOT be supported, because it's not supported by Kafka.
+The user will either need to revert the `spec.partitions`, or recreate the topic in Kafka (which they can do via the `KafkaTopic` or not).
+
+#### Changing `spec.replicas`
+
+This will NOT be supported as part of this proposal.
+
+#### Recreating topics in the Kafka cluster
+
+This WILL be supported.
```

Comment:
What exactly does this mean?

Reply:
Good question. I don't recall now what I meant by it. I'll remove it..